### PR TITLE
Migrate to modular github action in libbpf/ci repo.

### PR DESCRIPTION
**Pull request by thefallentree**

### CODE REVIEW

File: .github/actions/patch-kernel/action.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: 'patch kernel tree'
+description: 'patch kernel'
+runs:
+  using: "composite"
+  steps:
+    - name: patch bpf-next tree
```
Comment (anakryiko) [inline .github/actions/patch-kernel/action.yml:6]: we patch bpf tree as well? So maybe keep more generic: "apply temporary patches"?

File: .github/actions/vmtest/action.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+runs:
+  using: "composite"
+  steps:
+    # 1. Setup environment
+    - name: Setup build environment
+      uses: libbpf/ci/setup-build-env@master
+    # 2. Build
+    - name: Build kernel image
+      shell: bash
+      run: |
```
Comment (anakryiko) [inline .github/actions/vmtest/action.yml:12]: nit: this probably always going to be single shell script invocation, so "|" is unnecessary, right? We can keep it terse and single-line?

File: .github/actions/vmtest/build.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+set -euo pipefail
+
+THISDIR="$(cd $(dirname $0) && pwd)"
+
+source "${THISDIR}"/helpers.sh
+
+travis_fold start build_kernel "Building kernel"
+
+cp "${GITHUB_ACTION_PATH}"/latest.config .config
+make -j $((4*$(nproc))) olddefconfig all
```
Comment (anakryiko) [inline .github/actions/vmtest/build.sh:12]: /dev/null?

File: .github/actions/vmtest/build_selftests.sh [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+cd ${REPO_ROOT}/${REPO_PATH}
+make \
+	CLANG=clang-${LLVM_VER} \
+	LLC=llc-${LLVM_VER} \
+	LLVM_STRIP=llvm-strip-${LLVM_VER} \
+	VMLINUX_BTF="${VMLINUX_BTF}" \
+	VMLINUX_H="${VMLINUX_H}" \
+	-C "${REPO_ROOT}/${REPO_PATH}/tools/testing/selftests/bpf" \
+	-j $((4*$(nproc)))
```
Comment (anakryiko) [inline .github/actions/vmtest/build_selftests.sh:33]: /dev/null?

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -12,23 +13,24 @@ jobs:
...
         with:
-          repository: kernel-patches/bpf
-          ref: 'bpf-next'
-          path: 'linux'
-      - name: Move linux source in place
-        if: ${{ github.repository == 'kernel-patches/vmtest' }}
-        run: rsync --exclude .git -av linux/. . && rm -rf linux
-      - name: Kernel LATEST + selftests
-        run: export REPO_ROOT=$GITHUB_WORKSPACE; export CI_ROOT=$REPO_ROOT/travis-ci; export VMTEST_ROOT=$CI_ROOT/vmtest; ${CI_ROOT}/vmtest/run_vmtest.sh || exit 1
+          dest: '.kernel'
```
Comment (anakryiko) [inline .github/workflows/test.yml:26]: why not download right into the destination? did you run into any problems downloading sources to the final destination directory?

File: travis-ci/vmtest/run_selftests.sh [hunk 6]
```diff
@@ -7,15 +7,13 @@ source $(cd $(dirname $0) && pwd)/helpers.sh
...
-		echo "test_progs:$?" >> "${STATUS_FILE}"
-		travis_fold end test_progs
-	fi
+  travis_fold start test_progs "Testing test_progs"
+  # "&& true" does not change the return code (it is not executed
+  # if the Python script fails), but it prevents exiting on a
+  # failure due to the "set -e".
+  ./test_progs ${BLACKLIST:+-b$BLACKLIST} ${WHITELIST:+-t$WHITELIST} && true
+  echo "test_progs:$?" >> "${STATUS_FILE}"
+  travis_fold end test_progs
```
Comment (anakryiko) [inline travis-ci/vmtest/run_selftests.sh:16]: looks like indentation is off?

File: .github/workflows/test.yml (hunk 5 above)
Comment (thefallentree) [reply .github/workflows/test.yml:26]: Yeah, the script doesn't do well direct extracting to root directory, so for now i just left this as is

File: travis-ci/vmtest/run_selftests.sh (hunk 6 above)
Comment (thefallentree) [reply travis-ci/vmtest/run_selftests.sh:16]: github doesn't display tab characters well, I've switched everything back to space indent

**Pull request merged**